test/hotspot/jtreg/compiler/c2/aarch64/TestFarJump.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2022, BELLSOFT. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package compiler.c2.aarch64;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.*;
+
+/*
+ * @test
+ * @bug 8280872
+ * @summary Far call to runtime stub should be generated with single instruction for CodeHeap up to 250MB
+ * @library /test/lib /
+ *
+ * @requires vm.flagless
+ * @requires os.arch=="aarch64"
+ * @requires vm.compiler2.enabled
+ *
+ * @run driver compiler.c2.TestFarJump
```

Comment:
Package name `compiler.c2.aarch64` is different from `compiler.c2.TestFarJump`.
Got testing failure:

java.lang.ClassNotFoundException: compiler.c2.TestFarJump
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)

Reply:
After I fixed test name I got next failure:
```
[Exception Handler]
  0x0000fffe40140490: b3d5 9bd2 | 4201 80d2 | a4d5 9bd2 | a5d5 9bd2 

----------System.err:(11/622)----------
java.lang.RuntimeException: ADRP instruction is expected on far jump
	at compiler.c2.aarch64.TestFarJump.runVM(TestFarJump.java:112)
	at compiler.c2.aarch64.TestFarJump.main(TestFarJump.java:126)
```

Reply:
Thank you for checking that. My bad, last minute change have broken the test. I think the issue you observe is a debug build issue. I checked the functionality with branch_range=128M on debug build, and then reverted the fix:
- https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/assembler_aarch64.hpp#L854
```diff
  // The maximum range of a branch is fixed for the AArch64
  // architecture.  In debug mode we shrink it in order to test
  // trampolines, but not so small that branches in the interpreter
  // are out of range.
- static const uint64_t branch_range = NOT_DEBUG(128 * M) DEBUG_ONLY(2 * M);
+ static const uint64_t branch_range = 128 * M; // test only
```

In current change I disable the test for the debug build (``@requires vm.debug == false``)